torch/distributed/fsdp/flat_param.py

```diff
@@ -466,65 +499,131 @@ def _init_flat_param(
                         else param_name
                     )
                     fqns.append(fqn)
-        assert requires_grad is not None, (
-            "Passed-in `params` were not found in the module tree\n"
-            f"params: {params}\nmodule: {module}"
-        )
-        self.flat_param = FlatParamHandle.flatten_params(
-            params_to_flatten, requires_grad
+                    total_numel += param.numel()
+                    total_numel_without_padding += param.numel()
+        if len(params_to_flatten) == 0:
+            raise ValueError(
+                f"`params` were not found in `module`'s tree"
+                f"params: {params}\nmodule: {module}"
+            )
+        if (
+            self.rank == 0
+            and aligned_numel > 0
+            and total_numel != total_numel_without_padding
+        ):
+            logging.info(
```

Comment:
I will fix the lint related to this line soon, but I will wait for CI to finish running first.